pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1296,6 +1304,72 @@ func (p *PravegaCluster) ValidateSegmentStoreMemorySettings() error {
 	return nil
 }
 
+// ValidateBookkeeperSettings checks that the value passed for the options bookkeeper.ensemble.size (E) bookkeeper.write.quorum.size (W)
+// and bookkeeper.ack.quorum.size (A) adheres to the following rule, E >= W >= A.
+// The method also checks for the option bookkeeper.write.quorum.racks.minimumCount.enable which should be set to false when bookkeeper.ensemble.size is 1.
+// Note: The default value of E , W and A is 3.
+func (p *PravegaCluster) ValidateBookkeperSettings() error {
+	// Intializing ensemble size, write quorum size and ack quorum size to default value of 3
+	ensembleSizeInt, writeQuorumSizeInt, ackQuorumSizeInt := 3, 3, 3
+	var err error
+
+	ensembleSize := p.Spec.Pravega.Options["bookkeeper.ensemble.size"]
+	writeQuorumSize := p.Spec.Pravega.Options["bookkeeper.write.quorum.size"]
+	ackQuorumSize := p.Spec.Pravega.Options["bookkeeper.ack.quorum.size"]
+	writeQuorumRacks := p.Spec.Pravega.Options["bookkeeper.write.quorum.racks.minimumCount.enable"]
+
+	if len(ensembleSize) > 0 {
+		ensembleSizeInt, err = strconv.Atoi(ensembleSize)
+		if err != nil {
+			return fmt.Errorf("Cannot convert ensemble size from string to integer: %v", err)
+		}
+	}
+
+	if len(writeQuorumSize) > 0 {
+		writeQuorumSizeInt, err = strconv.Atoi(writeQuorumSize)
+		if err != nil {
+			return fmt.Errorf("Cannot convert write quorum size from string to integer: %v", err)
+		}
+	}
+
+	if len(ackQuorumSize) > 0 {
+		ackQuorumSizeInt, err = strconv.Atoi(ackQuorumSize)
+		if err != nil {
+			return fmt.Errorf("Cannot convert ack quorum size from string to integer: %v", err)
+		}
+	}
+
+	if writeQuorumRacks != "true" && writeQuorumRacks != "false" && writeQuorumRacks != "" {
+		return fmt.Errorf("bookkeeper.write.quorum.racks.minimumCount.enable can be only set to \"true\" \"false\" or \"\"")
+	}
+
+	if writeQuorumRacks == "true" && ensembleSizeInt == 1 {
+		return fmt.Errorf("bookkeeper.write.quorum.racks.minimumCount.enable should be set to false if bookkeeper.ensemble.size is 1")
+	}
+
+	if ensembleSizeInt < writeQuorumSizeInt {
+		if ensembleSize == "" {
+			return fmt.Errorf("The value provided for the option bookkeeper.write.quorum.size should be less than or equal to the default value of option bookkeeper.ensemble.size which is 3")
```

Comment:
The message is a bit confusing (the user might have forgotten to set `bookkeeper.ensemble.size` altogether).
Maybe change it to something like this: 
`"The value provided for the option bookkeeper.write.quorum.size should be less than or equal to the value of option bookkeeper.ensemble.size (default is 3)"`
Then the user would be aware of the need to **either** set the ensemble size explicitly **or** coordinate the quorum size with the default.